flake.nix

```diff
@@ -58,18 +63,35 @@
         };
       };
 
-      devShells.x86_64-linux.default = (project pkgsLocal).shellFor {
-        inherit (checks.x86_64-linux.pre-commit-check) shellHook;
-        withHoogle = false;
-        buildInputs =
-          with pkgsLocal;
-          [
-            haskellPackages.cabal-fmt
-            nixpkgs-fmt
-            nodejs
-          ]
-          ++ checks.x86_64-linux.pre-commit-check.enabledPackages;
-      };
+      devShells = forAllSystems (system:
+        let
+          pkgs = pkgsFor system;
+          proj = project pkgs;
+
+          # Only x86_64-linux has pre-commit checks defined
+          shellHook = if system == "x86_64-linux"
+                      then checks.x86_64-linux.pre-commit-check.shellHook
+                      else "";
```

Comment:
There's no reason to lock this to `x86_64-linux` only now that we're providing dev shells for other arch/os combinations. Use a `forAllSystems` to build the `checks` set and this conditional can go away.

And for future reference, this sort of condition exists in nixpkgs as [`lib.optionalString`](https://noogle.dev/f/lib/optionalString).